polar-core/src/vm.rs

```diff
@@ -1418,6 +1425,27 @@ impl PolarVirtualMachine {
                                 .collect::<Vec<Goals>>(),
                         )?;
                     }
+                    // Push an `ExternalLookup` goal for external instances and built-ins.
+                    Value::Dictionary(_) | Value::ExternalInstance(_) | Value::String(_) => {
+                        // Generate symbol for next result and bind to `false` (default)
+                        let next = self.kb.read().unwrap().gensym("next_value");
+                        self.bind(&next, Term::new_temporary(Value::Boolean(false)));
+                        let next_term = Term::new_temporary(Value::Variable(next.clone()));
+                        let call_id = self.new_call_id(&next);
+
```

Comment:
Kind of confused about exactly what's happening here, but I think this is just setting up a variable to receive the next value from the iterator?

Reply:
Yup! Same pattern we've used elsewhere. We might want to extract this into a helper method.

Reply:
did this `new_call_var`